### PR TITLE
add docs page for checks by importance

### LINKED ISSUE #361: [Documentation]: show checks by importance
### What would you like changed or added to the documentation and why?

It would be useful for the DANDI handbook to be able to link to a page in the NWB Inspector docs that list checks by importance. We should have headings for each importance level and then a list of each check with that importance. @CodyCBakerPhD can you think of a way to automate that?

You can get this organization with the following code:

```python
from collections import defaultdict

dd = defaultdict(list)

for x in available_checks:
    dd[x.importance].append(x)
```

Is there a way of automatically generating a markdown or rst file from this?

### Do you have any interest in helping write or edit the documentation?

No.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/nwbinspector/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this change was not already [requested](https://github.com/neurodatawithoutborders/nwbinspector/issues)?

### PR DESCRIPTION
**Pull request by bendichter**
fix #361

### CODE REVIEW

**Comment by bendichter** [comment]
I tried adding the docstrings of each check, but the 
```
Parameters
-----------
```
section was causes problems with .rst parsing so I removed them for now. I think it would be pretty straightforward to do some text manipulation to remove that part, but we can do that later.

File: docs/gen_checks_by_importance.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+This section lists the available checks organized by their importance level.
+
+''')
+
+        for importance_level, checks in dd.items():
+            f.write(f'''{importance_level}
+{'-' * (len(f'{importance_level}') + 1)}
+
+''')
```
Comment (CodyCBakerPhD) [inline docs/gen_checks_by_importance.py:26]: ?

Comment (bendichter) [reply docs/gen_checks_by_importance.py:26]: This is the title of each section

Comment (CodyCBakerPhD) [reply docs/gen_checks_by_importance.py:26]: Ah I see now - pre-commit must not be running on this, black usually as a different way of breaking the parentheses around triple quotes

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**